### DISCUSSION TITLE
Firefox extension client for mac unable to connect

### DISCUSSION THREAD

**Discussion by YannickVandereycken** [dani-garcia/vaultwarden#3838] [Q&A] [answered]
### Subject of the issue
When trying to connect or login via the firefox extension client for mac it throws an Network error:
![Screenshot 2023-09-03 at 15 19 02](https://github.com/dani-garcia/vaultwarden/assets/91282328/7626accb-d253-47cb-86b5-642b715a2949)
When doing this in firefox for windows, i don't have any problems.
The macOS app and iOS app also don't have any problems syncing.

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.29.2
* Web-vault version: v2023.7.1
* OS/Arch: linux/aarch64
* Running within Docker: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.41.2
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": true,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "",
  "domain_origin": "",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "push_enabled": false,
  "push_installation_id": "",
  "push_installation_key": "",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": true,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": "\"Plain\"",
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "",
  "smtp_password": "",
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 30,
  "smtp_username": "",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

* vaultwarden version: v1.29.2 

* Install method: Docker

* Clients used: firefox plugin client

* Reverse proxy and version: nginx proxy manager v2.10.4

* Other relevant details: 
- Firefox version 117.0
- Firefox extension version 2023.7.1

**Comment by BlackDex** [comment]
What log messages appear during the login attempts with the clients that can't seem to connect?

**Comment by YannickVandereycken** [comment]
where would i find these logs?

**Reply by YannickVandereycken** [reply]
after multiple tries i get this in the container logs:
[2023-09-03 13:52:01.790][vaultwarden::api::notifications][INFO] Closing WS connection from 192.168. .

**Reply by BlackDex** [reply]
Check you docker logs.
`docker logs -f vaultwarden`

**Reply by YannickVandereycken** [reply]
```
[2023-09-03 13:51:48.189][request][INFO] GET /api/config
[2023-09-03 13:51:48.190][response][INFO] (config) GET /api/config => 200 OK
[2023-09-03 13:52:01.790][vaultwarden::api::notifications][INFO] Closing WS connection from 192.168.*.*
[2023-09-03 13:53:55.470][request][INFO] GET /notifications/hub?access_token=********
[2023-09-03 13:53:55.470][vaultwarden::api::notifications][INFO] Accepting Rocket WS connection from 192.168.*.*
[2023-09-03 13:53:55.471][response][INFO] (websockets_hub) GET /notifications/hub?<data..> => 200 OK
[2023-09-03 13:53:55.486][request][INFO] GET /api/accounts/revision-date
[2023-09-03 13:53:55.489][response][INFO] (revision_date) GET /api/accounts/revision-date => 200 OK
[2023-09-03 13:54:00.228][vaultwarden::api::notifications][INFO] Closing WS connection from 192.168.*.*
```

**Reply by BlackDex** [reply]
Not seeing any errors.
Also, if other clients can access it without issues, it must be a client issue, or host/computer issue.
Vaultwarden does not differentiate between clients in any way.

Try to fully logout/uninstall the client/extension and reconfigure the client.

**Reply by YannickVandereycken** [reply]
i have already tried this, at one time it was successfully connected and then i noticed that a new password wasn't available in the firefox extension. I then tried to force sync but this didnt work so i searched around and then removed the extension restarted firefox en reinstalled it. But all of this was unsuccesful. I assumed this was because of the firefox version but then i noticed that the windows firefox version worked

**Reply by YannickVandereycken** [reply]
I redid de uninstallation/restart/installing the extension. But it didn't work.
I don't know if this detail is important, but it i don't have to logout since i can't do the login after first boot

**Comment by YannickVandereycken** [comment]
when trying to see if a wrong password would generate a log message, i did encounter another message which seems to have a missing variable connection
![Screenshot 2023-09-03 at 16 44 21](https://github.com/dani-garcia/vaultwarden/assets/91282328/f671fef9-5196-4c78-a72c-7a18aa28051d)

**Comment by BlackDex** [comment]
It has to be either something on the client side or something blocking the client on the server side.
Vaultwarden doesn't block, unless there is a rate-limit exceeded of login-attempts. But that works on IP base, and should then also prevent your other clients from trying to do a login.

If that is the case, you should see a message in the logs, which what you pasted wasn't the case.

The screenshot isn't a message which comes from Vaultwarden, but from the client, Vaultwarden doesn't know how long the password is the user typed in.

It could be some extensions, or other security software/toggles which cause this in some way.
But for that, i would suggest to check the logs of the client you are using. How to access to logs depends on the client. And i don't have any details on how to do that specifically per client.

**Reply by YannickVandereycken** [reply]
The strange thing is that the bitwarden macOS client does work on the same device. I don't know how differently this connects to the extension?

**Reply by BlackDex** [reply]
There isn't It should just make the exact same connection, nothing should be different.
But, the extensions can all interfere with each other.

I suggest to try a clean browser profile with no extensions other then Bitwarden and see if that solves it for you.

**Reply by YannickVandereycken** [reply]
using a clean profile worked

**Reply by YannickVandereycken** [reply]
but it is still strange that it would work on windows but not mac using the same logged in firefox profile

**Reply by YannickVandereycken** [reply, marked as answer]
I found the issue, it wasn't another extension but the firefox setting `DNS over HTTPS` set to Max Protection

**Reply by YannickVandereycken** [reply]
Thank you for helping me figure it out!
